src/rust/catnap/futures/connect.rs

```diff
@@ -68,21 +70,34 @@ impl Future for ConnectFuture {
     /// Polls the target [ConnectFuture].
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
         let self_: &mut ConnectFuture = self.get_mut();
-        match socket::connect(self_.fd as i32, &self_.addr) {
+        match unsafe {
+            libc::connect(
```

Comment:
I noticed that every libc invokation requires an unsafe block. Is that because these functions are unsafe? Are the nix ones safe somehow?

Reply:
Yes, invoking `libc` functions is unsafe. Functions from nix calls wrap around and provide a "safe" interface.